src/compiler/checker.ts

```diff
@@ -16325,12 +16325,35 @@ namespace ts {
             if (moduleSymbol) {
                 const esModuleSymbol = resolveESModuleSymbol(moduleSymbol, specifier, /*dontRecursivelyResolve*/ true);
                 if (esModuleSymbol) {
-                    return createPromiseReturnType(node, getTypeOfSymbol(esModuleSymbol));
+                    return createPromiseReturnType(node, syntheticDefaultType(getTypeOfSymbol(esModuleSymbol), esModuleSymbol));
                 }
             }
             return createPromiseReturnType(node, anyType);
         }
 
+        function syntheticDefaultType(type: Type, symbol: Symbol): Type {

```

Comment:
Can you use a more descriptive name like `getTypeWithSyntheticDefaultImport`? At a very high level this behaves similarly to `getTypeWithThisArgument` and `getTypeWithDefault` and as such should share a similar nomenclature.